packages/react-form/src/hooks/list/dynamiclist.ts

```diff
@@ -0,0 +1,74 @@
+import {useMemo, useEffect} from 'react';
+import isEqual from 'fast-deep-equal';
+
+import {NormalizedValidationDictionary, FieldDictionary} from '../../types';
+import {mapObject, normalizeValidation} from '../../utilities';
+
+import {
+  reinitializeAction,
+  useListReducer,
+  addFieldsAction,
+  removeFieldsAction,
+} from './reducer';
+import {FieldListConfig} from './list';
+import {useHandlers} from './hooks';
+
+interface DynamicList<Item extends object> {
+  fields: FieldDictionary<Item>[];
+  addField(): void;
+  removeField(index: number): void;
+}
+
+export function useDynamicList<Item extends object>(
+  listOrConfig: FieldListConfig<Item> | Item[],
+  fieldFactory: Function,
```

Comment:
Would like to see this function more tightly typed.
Also, should this be itemFactory? You use it to create an Item in the list that has multiple fields, correct?

Reply:
Yes, will change to itemFactory :) thanks!